services/FileAppend.proto

```diff
@@ -29,7 +29,7 @@ option java_multiple_files = true;
 import "BasicTypes.proto";
 
 /*
-Append the given contents to the end of the specified file. If a file is too big to create with a single FileCreateTransaction, then it can be created with the first part of its contents, and then appended as many times as necessary to create the entire file.
+Append the given contents to the end of the specified file. If a file is too big to create with a single FileCreateTransaction, then it can be created with the first part of its contents, and then appended as many times as necessary to create the entire file. This transaction must be signed by all initial M-of-M KeyList keys. If keys contains additional KeyList or ThresholdKey then M-of-M secondary KeyList or ThresholdKey signing requirements must be meet. 
```

Comment:
...M-of-M KeyList keys **associated with the file**

If **the** keys contain additional KeyList**s** or ThresholdKey**s,** ...

Reply:
"If the keys" = keys is being using to refer to the type KeyList keys = 3; not to a "key" in the generic sense